jota/src/test/java/org/iota/jota/IotaAPITest.java

```diff
@@ -78,384 +106,619 @@
+        GetNewAddressResponse newAddressResponse = GetNewAddressResponse
+                .create(Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_1), 497);
+
         AddressRequest addressRequest = new AddressRequest.Builder(TEST_SEED1, 1).checksum(true).amount(5).build();
-        final GetNewAddressResponse res1 = iotaAPI.getAddressesUnchecked(addressRequest);
-        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_1, res1.getAddresses().get(0));
 
-        AddressRequest secondAddressRequest = new AddressRequest.Builder(TEST_SEED1, 2).checksum(true).amount(5).build();
-        final GetNewAddressResponse res2 = iotaAPI.getAddressesUnchecked(secondAddressRequest);
-        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_2, res2.getAddresses().get(0));
+        when(iotaAPIMock.getAddressesUnchecked(addressRequest)).thenReturn(newAddressResponse);
 
-        AddressRequest thirdAddressRequest = new AddressRequest.Builder(TEST_SEED1, 3).checksum(true).amount(5).build();
-        final GetNewAddressResponse res3 = iotaAPI.getAddressesUnchecked(thirdAddressRequest);
-        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_3, res3.getAddresses().get(0));
+        GetNewAddressResponse addressResponse = iotaAPIMock.getAddressesUnchecked(addressRequest);
+        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_1, addressResponse.getAddresses().get(0));
     }
 
     @Test
-    public void shouldCreateANewAddressWithoutChecksum() throws ArgumentException {
+    public void shouldCreateNewAddressWithChecksumSecurityLevel2() throws ArgumentException {
+        GetNewAddressResponse newAddressResponse = GetNewAddressResponse
+                .create(Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_2), 497);
+
+        AddressRequest addressRequest = new AddressRequest.Builder(TEST_SEED1, 2).checksum(true).amount(5).build();
+
+        when(iotaAPIMock.getAddressesUnchecked(addressRequest)).thenReturn(newAddressResponse);
+
+        GetNewAddressResponse addressResponse = iotaAPIMock.getAddressesUnchecked(addressRequest);
+        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_2, addressResponse.getAddresses().get(0));
+    }
+
+    @Test
+    public void shouldCreateNewAddressWithChecksumSecurityLevel3() throws ArgumentException {
+        GetNewAddressResponse newAddressResponse = GetNewAddressResponse
+                .create(Collections.singletonList(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_3), 497);
+
+        AddressRequest addressRequest = new AddressRequest.Builder(TEST_SEED1, 3).checksum(true).amount(5).build();
+
+        when(iotaAPIMock.getAddressesUnchecked(addressRequest)).thenReturn(newAddressResponse);
+
+        GetNewAddressResponse res1 = iotaAPIMock.getAddressesUnchecked(addressRequest);
+        assertEquals(TEST_ADDRESS_WITH_CHECKSUM_SECURITY_LEVEL_3, res1.getAddresses().get(0));
```

Comment:
Codacy found an issue: [JUnit assertions should include a message](https://app.codacy.com/manual/Iota-java/iota-java/pullRequest?prid=6319461)